lib/pbio/drv/usb/stm32_usbd/usbd_pybricks.c

```diff
@@ -336,8 +323,8 @@ static USBD_StatusTypeDef USBD_Pybricks_Setup(USBD_HandleTypeDef *pdev,
   * @retval pointer to descriptor buffer
   */
 static uint8_t *USBD_Pybricks_GetCfgDesc(uint16_t *length) {
-    *length = (uint16_t)sizeof(USBD_Pybricks_CfgDesc);
-    return USBD_Pybricks_CfgDesc;
+    *length = (uint16_t)sizeof(USBD_Pybricks_CfgDesc.s);
```

Comment:
```suggestion
    *length = (uint16_t)sizeof(USBD_Pybricks_CfgDesc.s.conf_desc.wTotalLength);
```

Don't care too much either way, but it seems like this is the pattern we are using everywhere else.

Reply:
Although.... there are potentially a couple of issues with using the field instead of `sizeof()` in general.

1. Since it is 16-bit, it could cause unaligned access depending on how the struct is packed.
2. `sizeof()` is guaranteed to be a compile time constant while the field doesn't offer guarantees. The compiler _should_ see that it is a constant and generate the same code, but sometimes compilers aren't so smart.

So if using `sizeof()` everywhere generates smaller firmware binary on NXT, we should go with that.

Reply:
Oh, except it *still* doesn't work if there's a VLA involved, which currently only affects the WebUSB descriptor.